src/components/screens/ValidateRecoveryPhraseScreen.tsx

```diff
@@ -22,62 +31,87 @@ export const ValidateRecoveryPhraseScreen: React.FC<
   ValidateRecoveryPhraseScreenProps
 > = ({ route, navigation }) => {
   const { password, recoveryPhrase } = route.params;
-  const [currentWord, setCurrentWord] = useState("");
+  const [selectedWord, setSelectedWord] = useState<string>("");
   const [error, setError] = useState<string | undefined>();
   const [isLoading, setIsLoading] = useState(false);
   const isSigningUp = useAuthenticationStore((state) => state.isLoading);
-  const [currentIndex, setCurrentIndex] = useState(0);
+  const [roundIndex, setRoundIndex] = useState(0);
+  const [isRegeneratingWords, setIsRegeneratingWords] = useState(false);
 
   const { signUp } = useAuthenticationStore();
   const { t } = useAppTranslation();
+  const { themeColors } = useColors();
+
+  const { words, selectedIndexes, generateWordOptionsForRound } =
+    useWordSelection(recoveryPhrase);
+  const currentWordIndex = selectedIndexes[roundIndex];
+  const currentCorrectWord = words[currentWordIndex];
+
+  const [currentRoundWordOptions, setCurrentRoundWordOptions] = useState<
+    string[]
+  >([]);
+
+  const regenerateWords = useCallback(() => {
+    setIsRegeneratingWords(true);
+
+    const newWordOptions = generateWordOptionsForRound(roundIndex);
+    setCurrentRoundWordOptions(newWordOptions);
+
+    setIsRegeneratingWords(false);
+    setSelectedWord("");
```

Comment:
just doing a sanity check: do we need to `setIsRegeneratingWords` even though there is no async operation happening here?